src/dom.js

```diff
@@ -239,6 +239,23 @@ export function lastChildElement(parent, callback) {
   return null;
 }
 
+/**
+ * Finds all child nodes that satisfies the callback.
+ * These nodes can include Text, Comment and other child nodes.
+ * @param {!Node} parent
+ * @param {function(!Node):boolean} callback
+ * @return {!Array<!Node>}
+ */
+export function childNodes(parent, callback) {
+  const childnodes = [];
```

Comment:
nit: call this `children` or `nodes`. `childnodes` doesn't follow the camel-case style.